src/rsvc/srv.c

```diff
@@ -1152,21 +1202,22 @@ bcast_create(crt_opcode_t opc, bool filter_invert, d_rank_list_t *filter_ranks,
 }
 
 /**
- * Perform a distributed create, if \a create is true, and start operation on
- * all replicas of a database with \a dbid spanning \a ranks. This method can
- * be called on any rank. If \a create is false, \a ranks may be NULL.
+ * Perform a distributed start operation in \a mode on all replicas of a
+ * database with \a dbid spanning \a ranks. This method can be called on any
+ * rank. If \a mode is DS_RSVC_START, \a ranks may be NULL. If \a mode is
+ * DS_RSVC_DICTATE, \a ranks must comprise one and only one rank.
  *
  * \param[in]	class		replicated service class
  * \param[in]	id		replicated service ID
  * \param[in]	dbid		database UUID
  * \param[in]	ranks		list of replica ranks
- * \param[in]	create		create replicas first
- * \param[in]	bootstrap	start with an initial list of replicas
- * \param[in]	size		size of each replica in bytes if \a create
+ * \param[in]	mode		mode of starting the replicated service
+ * \param[in]	bootstrap	create with an initial list of replicas if \a mode is DS_RSVC_CREATE
+ * \param[in]	size		size of each replica in bytes if \a mode is DS_RSVC_CREATE
  */
 int
 ds_rsvc_dist_start(enum ds_rsvc_class_id class, d_iov_t *id, const uuid_t dbid,
-		   const d_rank_list_t *ranks, bool create, bool bootstrap,
+		   const d_rank_list_t *ranks, enum ds_rsvc_start_mode mode, bool bootstrap,
```

Comment:
Consider the following scenario: originally the pool have 3 PS replicas, assume they are on rank_0/1/2. Here if we specify the rank_0/1 to start the PS, please note that rank_2 is not dead, just not start the PS, then after PS started, the raft mechanism will automatically generate another PS replica on rank_x? or can try to re-add rank_2 to the replicas list?

Reply:
The architecture looks like this:

    pool_svc module and chk_leader module // decide when and how to add/remove PS replicas
    ds_rsvc module
    rdb module
    raft module

Only the pool_svc module and the chk_leader module initiate replica addition/removal actions. In the scenario above, why do "we specify the rank_0/1 to start the PS"?

Reply:
In this example, rank_0 thinks rank_0/1 are valid replicas, rank_1 thinks rank0/1 are valid replicas, only rank2 things rank_0/1/2 are valid replicas, but from the check leader's perspective, rank_2 is the lost one in the scan list. When it scan the first replica in the list, it found that rank_0 has majority replicas to vote to rank_0, then check leader will think the PS is available, under such, if the check leader wants to start PS for this pool, it will specify the rank_list based on rank_0's perspective via ds_rsvc_dist_start(rank0/1).

Reply:
It is not under dictate mode for this example.

Reply:
As we discussed offline, we'll try one of two approaches in the chk leader:

- Start replicas on replica ranks, that is, if no dictate operation, then rank 0, 1, and 2, otherwise just the chosen rank.
- Start all replicas on all ranks, that is, in a similar way as start_one in src/pool/srv_pool.c. This will make the process similar to normal pool start.